File: go/etl/etl.go
package etl

import (
	"strings"
)

func Transform(in map[int][]string) map[string]int {
	var result = make(map[string]int)

	for score, letters := range in {
		for _, letter := range letters {
			result[strings.ToLower(letter)] = score
		}
	}

	return result
}

File: typescript/etl/etl.ts
interface OldFormat {
    [key: string]: string[];
}

interface NewFormat {
    [key: string]: number;
}

export function transform(old: OldFormat): NewFormat {
    const result: NewFormat = {};

    for (const pointValue in old) {
        const letters = old[pointValue];

        for (const letter of letters) {
            result[letter.toLowerCase()] = parseInt(pointValue, 10);
        }
    }

    return result;
}
